src/site/content/en/blog/progressively-enhance-your-pwa/index.md

```diff
@@ -0,0 +1,1051 @@
+       src="10000201000009C4000005DBCD9D10DD0A745315.png"
+       alt="The insomnia checkbox, if checked, keeps the screen awake.">
+  <figcaption class="w-figcaption">
+    The insomnia checkbox keeps app awake.
+  </figcaption>
+</figure>
+
+## The Idle Detection API
+
+At times, even if you stare at the screen for hours,
+it's just useless.
+The [Idle Detection API](https://web.dev/idle-detection/) allows the app to detect user idle time.
+If the user is idle for too long, the app resets to the initial state
+and clears the canvas.
+This API is currently gated behind the
+[notifications permission](https://developer.mozilla.org/en-US/docs/Web/API/Notification/requestPermission),
+since a lot of production use cases of idle detection are notifications-related,
+for example, to only send a notification to a device the user is currently actively using.
+
+After making sure that the notifications permission is granted, I then instantiate the
+idle detector.
+I register an event listener that listens for idle changes, which includes the user and
+the screen state.
+The user can be active or idle,
+and the screen can be unlocked or locked.
+If the user is idle, the canvas clears.
+I give the idle detector a threshold of 60 seconds.
+
+```js
+const idleDetector = new IdleDetector();
+idleDetector.addEventListener('change', () => {
+  const userState = idleDetector.userState;
+  const screenState = idleDetector.screenState;
+  console.log(`Idle change: ${userState}, ${screenState}.`);
+  if (userState === 'idle') {
+    clearCanvas();
+  }
+});
+
+await idleDetector.start({
+  threshold: 60000,
+  signal,
+});
+```
+
+And as always, I only load this code when the browser supports it.
+
+```js
+if ('IdleDetector' in window) {
+  import('./idle_detection.mjs');
+}
+```
+
+In the Fugu Greetings app, the canvas clears when the Ephemeral checkbox is
+checked and the user is idle for for too long.
+
+<figure class="w-figure">
+  <img class="w-screenshot"
+       src="10000201000009C4000005DBB930B1D3F8F5D1DA.png"
+       alt="Fugu Greetings app with a cleared canvas after the user has been idle for too long.">
+  <figcaption class="w-figcaption">
+    When the ephemeral checkbox is checked and the user has been idle for too long, the canvas is cleared.
+  </figcaption>
+</figure>
+
+## Closing
+
+Phew, what a ride. So many APIs in just one sample app.
+And, remember, I never make the user pay the download cost
+for a feature that their browser doesn't support.
+By using progressive enhancement, I make sure only the relevant code gets loaded.
+And since with HTTP/2, requests are cheap, this pattern should work well for a lot of
+applications,
+although you might want to consider a bundler for really large apps.
+
+<figure class="w-figure">
+  <img class="w-screenshot"
+       src="10000201000009C4000006E8D776365D6B538C9A.png"
+       alt="Chrome DevTools Network tab showing only requests for files with code that the current browser supports.">
```

Comment:
```suggestion
       alt="Chrome DevTools Network panel showing only requests for files with code that the current browser supports.">
```
In the official docs I always refer to the top-level UI containers as `panels`. I believe the phrase predated me. I know `tab` is a more intuitive term but it would be nice to keep the alignment with the official docs terminology throughout this post.